go/api/command_options.go

```diff
@@ -279,6 +280,128 @@ func (listDirection ListDirection) toString() (string, error) {
 	}
 }
 
+const (
+	// LIMIT subcommand string to include in the SORT and SORT_RO commands.
+	LIMIT_COMMAND_STRING = "LIMIT"
+	// ALPHA subcommand string to include in the SORT and SORT_RO commands.
+	ALPHA_COMMAND_STRING = "ALPHA"
+	// BY subcommand string to include in the SORT and SORT_RO commands.
+	// Supported in cluster mode since Valkey version 8.0 and above.
+	BY_COMMAND_STRING = "BY"
+	// GET subcommand string to include in the SORT and SORT_RO commands.
+	GET_COMMAND_STRING = "GET"
+)
+
+// Limit struct represents the range of elements to retrieve
+// The LIMIT argument is commonly used to specify a subset of results from the matching elements, similar to the
+// LIMIT clause in SQL (e.g., `SELECT LIMIT offset, count`).
+type Limit struct {
+	// The starting position of the range, zero based.
+	Offset int64
+	// The maximum number of elements to include in the range. A negative count returns all elementsnfrom the offset.
+	Count int64
+}
+
+// OrderBy specifies the order to sort the elements. Can be ASC (ascending) or DESC(descending).
+type OrderBy string
+
+const (
+	ASC  OrderBy = "ASC"
+	DESC OrderBy = "DESC"
+)
+
+// SortOptions struct combines both the base options and additional sorting options
+type SortOptions struct {
+	// Limit Limits the range of elements
+	Limit *Limit
+
+	// OrderBy sets the order to sort by (ASC or DESC)
+	OrderBy OrderBy
+
+	// IsAlpha determines whether to sort lexicographically (true) or numerically (false)
+	IsAlpha bool
+
+	// ByPattern - a pattern to sort by external keys instead of by the elements stored at the key themselves. The
+	// pattern should contain an asterisk (*) as a placeholder for the element values, where the value
+	// from the key replaces the asterisk to create the key name. For example, if key
+	// contains IDs of objects, byPattern can be used to sort these IDs based on an
+	// attribute of the objects, like their weights or timestamps. Supported in cluster mode since
+	// Valkey version 8.0 and above.
```

Comment:
```suggestion
	// attribute of the objects, like their weights or timestamps.
	// Supported in cluster mode since Valkey version 8.0 and above.
```